uaaextras/webapp.py

```diff
@@ -383,6 +394,93 @@ def oauth_login():
     def index():
         return render_template('index.html')
 
+    @app.route('/signup', methods=['GET', 'POST'])
```

Comment:
This all makes sense, but could it be split into a few smaller functions? Maybe `is_valid_email`, `ensure_uaa_client`. Also, this is beyond the scope of the work happening here, but could we pass the uaa client credentials to `UAAClient` so that we don't have to pass them to all of its methods?